server/util/file.py

```diff
@@ -74,6 +84,24 @@ def retrieve_file(storage_path: str) -> BinaryIO:
         return open(storage_path, "rb")
 
 
+def retrieve_file_streaming(storage_path: str, working_directory: str) -> BinaryIO:
+    if config.FILE_UPLOAD_STORAGE_PATH.startswith("s3://"):
+        assert storage_path.startswith(config.FILE_UPLOAD_STORAGE_PATH)
+        parsed_path = urlparse(storage_path)
+        bucket_name = parsed_path.netloc
+        key = parsed_path.path[1:]
+        temp_file_path = os.path.join(working_directory, key)
```

Comment:
I strongly recommend not using the `path` from the s3 URL if there's any chance we don't fully control it.